Compile error with CONFIG+="headless": src/util.cpp: variable ‘QRegularExpression rx1’ has initializer but incomplete type

**Describe the bug**
<!-- A clear and concise description of what the bug is. -->
Report by @corrados as part of #2267:
> Unforunately, the current Jamulus source code does not compile on my Raspberry Pi with the following error message:

**To Reproduce**
<!-- Steps to reproduce the behavior: -->
- `distributions/raspijamulus.sh`
- Any current build with `CONFIG+="headless"`

**Expected behavior**
<!-- A clear and concise description of what you expected to happen. -->
Sucessful compile

**Screenshots**
<!-- If applicable, add screenshots to help explain your problem. -->
```
src/util.cpp: In static member function ‘static bool NetworkUtil::ParseNetworkAddress(QString, CHostAddress&, bool)’:
src/util.cpp:708:28: error: variable ‘QRegularExpression rx1’ has initializer but incomplete type
  708 |     QRegularExpression rx1 ( "^\\[([^]]*)\\](?::(\\d+))?$" ); // [addr4or6] or [addr4or6]:port
      |                            ^
src/util.cpp:709:28: error: variable ‘QRegularExpression rx2’ has initializer but incomplete type
  709 |     QRegularExpression rx2 ( "^([^:]*)(?::(\\d+))?$" );       // addr4 or addr4:port or host or host:port
      |                            ^
src/util.cpp:713:29: error: variable ‘QRegularExpressionMatch rx1match’ has initializer but incomplete type
  713 |     QRegularExpressionMatch rx1match = rx1.match ( strAddress );
      |                             ^~~~~~~~
src/util.cpp:714:29: error: variable ‘QRegularExpressionMatch rx2match’ has initializer but incomplete type
  714 |     QRegularExpressionMatch rx2match = rx2.match ( strAddress );
      |                             ^~~~~~~~
make: *** [Makefile:1762: util.o] Error 1
```

**Operating system**
<!-- Add the operating system you are running, e.g. Windows 10 Version 20H2, macOS 10.15.5, Ubuntu Linux 20.10. Also add where you got Jamulus from, e.g. if you downloaded it from the internet, if you got it from your Linux distribution or if you built from source. -->
Latest Raspbian (?)

**Version of Jamulus**
<!-- Get this from the Help > About menu or from the `Jamulus --version` command line -->
`master`

**Additional context**
<!-- Add any other context about the problem here. -->

--- a/ChangeLog
+++ b/ChangeLog
@@ -28,8 +28,9 @@
   (contributed by @ngocdh)
 
 - Client: Improved regex to highlight URLs in the chat window, avoiding terminating punctuation.
-  Also migrated from deprecated QRegExp to QRegularExpression, for future compatibility with Qt6 (#2124).
-  (contributed by @softins)
+  Also migrated from deprecated QRegExp to QRegularExpression, for future compatibility with
+  Qt6 (#2124, #2272, #2273).
+  (contributed by @softins, @corrados, @hoffie)
 
 - Client: Improved keyboard control of the list of Custom Directories (#2129).
   (contributed by @pljones)

--- a/src/chatdlg.h
+++ b/src/chatdlg.h
@@ -34,6 +34,7 @@
 #include <QAccessible>
 #include <QDesktopServices>
 #include <QMessageBox>
+#include <QRegularExpression>
 #include "global.h"
 #include "util.h"
 #include "ui_chatdlgbase.h"

--- a/src/connectdlg.h
+++ b/src/connectdlg.h
@@ -31,6 +31,7 @@
 #include <QTimer>
 #include <QLocale>
 #include <QtConcurrent>
+#include <QRegularExpression>
 #include "global.h"
 #include "util.h"
 #include "settings.h"

--- a/src/util.h
+++ b/src/util.h
@@ -45,6 +45,7 @@
 #endif
 #include <QFile>
 #include <QDirIterator>
+#include <QRegularExpression>
 #include <QTranslator>
 #include <QLibraryInfo>
 #include <QUrl>

